### PR TITLE
[API Deprecation]Deprecate candidates in dgl.distributed module

### PR DESCRIPTION
**Pull request by peizhou001**
## Description

1. Remove shuffle argument in partition_book function.
2. Remove Class BasicPartitionBook.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by mufeili** [comment]
This must get approved by @Rhett-Ying and (@zheng-da or @classicsong ).

File: python/dgl/distributed/partition.py
Comment (Rhett-Ying) [inline python/dgl/distributed/partition.py:339]: how about moving raise exception before handle: 
```
if not is_range_part: raise
#handle RangePartitionBook
```

File: python/dgl/distributed/partition.py
Comment (Rhett-Ying) [inline python/dgl/distributed/partition.py:731]: why not remove the `reshuffle` argument in `_get_orig_ids()`? blocked by someone else uses this function?

File: tests/distributed/test_mp_dataloader.py
Comment (Rhett-Ying) [inline tests/distributed/test_mp_dataloader.py:344]: seems forgot to remove `@pytest.mark.parametrize("reshuffle", [True, False])` in the test func

File: tests/distributed/test_new_kvstore.py
Comment (Rhett-Ying) [inline tests/distributed/test_new_kvstore.py:21]: how about keep the original node/edge_map and the new one in test. then we could make sure original one will raise exception and the new one will work well.

Comment (Rhett-Ying) [reply tests/distributed/test_new_kvstore.py:21]: we need to make sure `is_range_part = False` cases could be handled: raise exception.

File: python/dgl/distributed/partition.py
Comment (peizhou001) [reply python/dgl/distributed/partition.py:339]: sure, move for fix.

File: python/dgl/distributed/partition.py
Comment (peizhou001) [reply python/dgl/distributed/partition.py:731]: Line 690 set `reshuffle` to False, will remove this argument influence the normal behavior?

File: tests/distributed/test_mp_dataloader.py
Comment (peizhou001) [reply tests/distributed/test_mp_dataloader.py:344]: Nice catch! Removed for fix

File: tests/distributed/test_new_kvstore.py
Comment (peizhou001) [reply tests/distributed/test_new_kvstore.py:21]: Got you, but how about adding a new test case in `test_partition`? which should be more suitable.

Comment (peizhou001) [reply tests/distributed/test_new_kvstore.py:21]: Got you, but keep the original node/map here will not cover the `is_range_part`, let me add a new test in test_partition.

File: python/dgl/distributed/graph_partition_book.py
Comment (frozenbugs) [inline python/dgl/distributed/graph_partition_book.py:277]: 80 char max.

File: python/dgl/distributed/partition.py
Comment (Rhett-Ying) [reply python/dgl/distributed/partition.py:731]: oh, but leaves an argument named `reshuffle` looks a bit weird. we're supposed to remove all `reshuffle`. could you check the difference between `reshuffle=True` and `reshuffle=False` in this case? we need to try to remove it or we need add comments to clarify why we need this argument. what do you think

File: python/dgl/distributed/graph_partition_book.py
Comment (peizhou001) [reply python/dgl/distributed/graph_partition_book.py:277]: fixed

File: python/dgl/distributed/partition.py
Comment (peizhou001) [reply python/dgl/distributed/partition.py]: removed for fix

File: tests/distributed/test_new_kvstore.py
Comment (Rhett-Ying) [reply tests/distributed/test_new_kvstore.py:21]: have you added some tests to cover the exception code path?

Comment (peizhou001) [reply tests/distributed/test_new_kvstore.py:21]: see test_partition

**Comment by frozenbugs** [comment]
> This must get approved by @Rhett-Ying and (@zheng-da or @classicsong ).

LGTM, please get approval by them and merge without waiting for my approval.

**Review by Rhett-Ying — APPROVED** [review]
LGTM

**Pull request merged**
